### PR TITLE
feature(scale 2D): add scale on 2D views

### PR DESCRIPTION
**Pull request by mgermerie**
## Description

Add a scale div in view_2d_map example.
Add an event triggering in PlanarControls when a view animation is over.
Detect this event in view_2d_map example to compute scale.

### CODE REVIEW

**Comment by gchoqueux** [comment]
Could you add scale widget on 25d example?
`Travel` isn't the only move to change the range.
In this example, you can try other moves.

**Comment by mgermerie** [comment]
I added a scale on the 2.5D map example.
Further more, a `RANGE_CHANGED` event is now also triggered when user performs a `pan` movement.

File: examples/view_25d_map.html [hunk 1]
```diff
@@ -97,6 +99,33 @@
             // eslint-disable-next-line no-new
             new itowns.PlanarControls(view, {});
 
+            // Add scale :
+            let divScaleWidget = document.getElementById('divScaleWidget');
+            function updateScaleWidget() {
```
Comment (gchoqueux) [inline examples/view_25d_map.html:106]: Could you factorize `updateScaleWidget` in `examples/js ` to share the function

File: src/Controls/PlanarControls.js [hunk 2]
```diff
@@ -219,6 +223,9 @@ function PlanarControls(view, options = {}) {
         }
         if (this.state === STATE.TRAVEL) {
             this.handleTravel(dt);
+            this.view.dispatchEvent({
+                type: PLANAR_CONTROL_EVENT.RANGE_CHANGED,
+            });
             this.view.notifyChange(this.camera);
         }
         if (this.state === STATE.DRAG) {
```
Comment (gchoqueux) [inline src/Controls/PlanarControls.js]: I think for all `STATE` the range will change

File: examples/js/Scale.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/**
+ * For a given pixel distance at the center of a given view, calculates the equivalent metric distance.
```
Comment (gchoqueux) [inline examples/js/Scale.js]: there's not pixel distance from center, it's projected pixels on ground.

File: examples/view_25d_map.html [hunk 4]
```diff
@@ -97,6 +100,25 @@
             // eslint-disable-next-line no-new
             new itowns.PlanarControls(view, {});
 
+            // Initialize scale :
+            let initialScaleSize = 200;
```
Comment (gchoqueux) [inline examples/view_25d_map.html]: it's constant

**Review by gchoqueux — COMMENTED** [review]
Could squash all fix commits?

It's feature, not a fix, no?

File: examples/js/Scale.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+/**
+ * For a given pixel size, calculates the projected metric distance at the middle of the screen.
+ * Calculates then the pixel distance equivalent to the rounded computed metric distance.
+ *
+ * @param {itowns.View} view View in which the distances are computed.
+ * @param {number} distancePixels Distance in pixels, at the center of the view.
+ * @return {{pixels: number, meters: number}} The rounded pixel and metric distances.
+ */
+// eslint-disable-next-line no-unused-vars
+function roundPixelsFromMeters(view, distancePixels) {
```
Comment (gchoqueux) [inline examples/js/Scale.js]: it's not distancePixels

File: examples/view_25d_map.html [hunk 6]
```diff
@@ -97,6 +100,25 @@
             // eslint-disable-next-line no-new
             new itowns.PlanarControls(view, {});
 
+            // Initialize scale :
+            const initialScaleSize = 200;
```
Comment (gchoqueux) [inline examples/view_25d_map.html]: could you add `in pixel`

**Review by gchoqueux — APPROVED** [review]

**Comment by gchoqueux** [comment]
could you rebase on master?

**Pull request merged**